sentry-android-core/src/main/java/io/sentry/android/core/SentryAndroidDateProvider.java

```diff
@@ -0,0 +1,32 @@
+package io.sentry.android.core;
+
+import android.os.Build;
+import io.sentry.SentryDate;
+import io.sentry.SentryDateProvider;
+import io.sentry.SentryInstantDateProvider;
+import io.sentry.SentryNanotimeDateProvider;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This {@link SentryDateProvider} will use {@link java.time.Instant} for Android API 26+ or a
+ * combination of {@link java.util.Date} and System.nanoTime() on lower versions.
+ */
+@ApiStatus.Internal
+public final class SentryAndroidDateProvider implements SentryDateProvider {
+
+  private final @NotNull SentryDateProvider dateProvider;
+
+  public SentryAndroidDateProvider(final @NotNull BuildInfoProvider buildInfoProvider) {
+    if (buildInfoProvider.getSdkInfoVersion() >= Build.VERSION_CODES.O) {
+      dateProvider = new SentryInstantDateProvider();
+    } else {
```

Comment:
`h`: as I explained down in the comments, we could still try to access `Instant.now` with a try-catch, and fallback to `SentryNanotimeDateProvider` in the catch block.